config/configcompression/compressiontype.go

```diff
@@ -40,3 +50,25 @@ func (ct *Type) UnmarshalText(in []byte) error {
 	}
 	return fmt.Errorf("unsupported compression type %q", typ)
 }
+
+func (t Type) ValidateParams(p CompressionParams) error {
+	switch t {
+	case TypeGzip, TypeZlib, TypeDeflate:
+		if p.Level == zlib.DefaultCompression ||
+			p.Level == zlib.HuffmanOnly ||
+			p.Level == zlib.NoCompression ||
+			(p.Level >= zlib.BestSpeed && p.Level <= zlib.BestCompression) {
+			return nil
+		}
+	case TypeZstd:
+		// Supports arbitrary levels: zstd will map any given
+		// level to the nearest internally supported level.
+		return nil
+	case TypeSnappy, TypeLz4, typeNone, typeEmpty:
+		if p.Level != 0 {
+			return fmt.Errorf("unsupported parameters %+v for compression type %q", p, t)
+		}
+		return nil
+	}
+	return fmt.Errorf("unsupported parameters %+v for compression type %q", p, t)
```

Comment:
Do we need to list all the types explicitly here? It'll be hard to keep it in sync when new types are added. What about this instead?
```suggestion
	if p.Level != 0 {
		return fmt.Errorf("unsupported parameters %+v for compression type %q", p, t)
	}
	return nil
```

Reply:
Maybe this is fine? We'll need to revisit this function when new types are added anyway...

Reply:
on second thought, it seems unnecessary, made the update, thanks